### PR TITLE
Feature/add/inline consent dialog to front page

### PR DESCRIPTION
**Pull request by ag12**
# What?
New Dialogue Component 
And Inline Consent.

### CODE REVIEW

File: Demo/Extensions/UIKit/UIDevice.swift [hunk 1]
```diff
@@ -6,7 +6,7 @@ import UIKit.UIDevice
 
 // MARK: - UIDevice
 
- extension UIDevice {
+ internal extension UIDevice {
```
Comment (vadymmarkov) [inline Demo/Extensions/UIKit/UIDevice.swift]: It should be `internal` by default.

File: Sources/Extensions/UIKit/UIDevice.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
+//
+//  Copyright © 2019 FINN AS. All rights reserved.
+//
+
+import UIKit.UIDevice
+
+// MARK: - UIDevice
+
+internal extension UIDevice {
```
Comment (vadymmarkov) [inline Sources/Extensions/UIKit/UIDevice.swift]: The same comment about `internal` as above.

File: Sources/Fullscreen/FrontPage/FrontPageView.swift [hunk 3]
```diff
@@ -185,28 +186,51 @@ public final class FrontPageView: UIView {
...
         adsRetryView.frame.origin = CGPoint(x: 0, y: headerView.frame.height + .veryLargeSpacing)
         adsRetryView.frame.size = CGSize(width: bounds.width, height: 200)
         boundsForCurrentSubviewSetup = bounds
         adsGridView.invalidateLayout()
+        setupFrameForDialogue(height: height)
     }
 
-    private func updateHeaderTitle() {
-        headerLabel.text = inlineConsentView.isHidden ? model?.adsGridViewHeaderTitle : model?.inlineConsentTitle
+    func setupFrameForDialogue(height: CGFloat) {
```
Comment (vadymmarkov) [inline Sources/Fullscreen/FrontPage/FrontPageView.swift]: Maybe `y` or `yPosition` is better seeing you don't use it as a height.

File: Sources/Extensions/UIKit/UIDevice.swift [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+//  Copyright © 2019 FINN AS. All rights reserved.
+//
+
+import UIKit.UIDevice
+
+// MARK: - UIDevice
+
+extension UIDevice {
+
+    internal class func isIPad() -> Bool {
```
Comment (vadymmarkov) [inline Sources/Extensions/UIKit/UIDevice.swift]: You can also remove `internal` from class function here.

File: Sources/Components/Dialogue/DialogueView.swift [hunk 5]
```diff
@@ -147,25 +138,42 @@ public class DialogueView: UIView {
...
+            link.topAnchor.constraint(equalTo: primaryButton.bottomAnchor, constant: .mediumSpacing),
             link.leadingAnchor.constraint(equalTo: leadingAnchor, constant: .largeSpacing),
             link.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -.largeSpacing),
-
-            primaryButton.topAnchor.constraint(equalTo: link.bottomAnchor, constant: .mediumLargeSpacing),
-            primaryButton.centerXAnchor.constraint(equalTo: centerXAnchor),
             ])
     }
 
+    func heightWithConstrainedWidth(width: CGFloat) -> CGFloat {
```
Comment (pimms) [inline Sources/Components/Dialogue/DialogueView.swift]: What about `heightWithConstrainedWidth(_ width: CGFloat)` instead?

File: Sources/Components/Dialogue/DialogueView.swift [hunk 6]
```diff
@@ -147,25 +138,42 @@ public class DialogueView: UIView {
...
+        let constraintRect = CGSize(width: width, height: .greatestFiniteMagnitude)
+        let boundingBox = text.boundingRect(with: constraintRect, options: [.usesLineFragmentOrigin], attributes: [NSAttributedString.Key.font: detail.font], context: nil)
+
+        var moreSpacing: CGFloat = 0
+
+        if UIDevice.isSmallScreen() {
+            moreSpacing = .largeSpacing
+        }
+
+        if UIDevice.isLandscape() {
```
Comment (pimms) [inline Sources/Components/Dialogue/DialogueView.swift]: Should this be an `else if UIDevice.isLandscape` instead? It looks like they are exclusive.

File: Sources/Extensions/UIKit/UIViewExtensions.swift [hunk 7]
```diff
@@ -33,4 +33,13 @@ public extension UIView {
             return layoutMarginsGuide
         }
     }
+
+    public func dropShadow(color: UIColor, opacity: Float = 0.5, offset: CGSize = CGSize.zero, radius: CGFloat = 10.0, usingPath: Bool = true) {
```
Comment (pimms) [inline Sources/Extensions/UIKit/UIViewExtensions.swift]: `usingPath` is not used

**Comment by ag12** [comment]
Thanks for the feedback @finn-no/ios

**Pull request merged**